react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.createTitle = this.createTitle.bind(this);
+    this.state = { clickedTitle: null };
+  }
+
+  handleClick() {
+    event.target.id === this.state.clickedTitle
+      ? this.setState({ clickedTitle: null })
+      : this.setState({ clickedTitle: event.target.id });
+  }
+
+  createDescription(id) {
+    for (let i = 0; i < this.props.shows.length; i++) {
+      if (this.props.shows[i][id]) return <p className="description">{this.props.shows[i][id]}</p>;
+    }
+  }
+
+  createTitle() {
+    const frag = <></>;
+    frag.props.children = [];
+    for (let i = 0; i < this.props.shows.length; i++) {
+      Object.keys(this.props.shows[i])[0] === this.state.clickedTitle
+        ? frag.props.children.push(<h1 className="title" id={Object.keys(this.props.shows[i])} onClick={this.handleClick}>{Object.keys(this.props.shows[i])}</h1>, this.createDescription(this.state.clickedTitle))
+        : frag.props.children.push(<h1 className="title" id={Object.keys(this.props.shows[i])} onClick={this.handleClick}>{Object.keys(this.props.shows[i])}</h1>);
+    }
+    return frag;
+  }
```

Comment:
In React, we try to stick with `Array.prototype.map` when we generate an array of React elements. You can include conditional logic in your `.map()` callback to return something different 

```jsx
const children = this.props.shows.map(show => {
  if (show.title === this.state.clickedTitle) {
    return <something />
  } else {
    return <something-else />
})
return (
  <>
    { children }
  </>
```